components/trusted-app-mgt/org.wso2.carbon.identity.trusted.app.mgt/src/main/java/org/wso2/carbon/identity/trusted/app/mgt/services/TrustedAppMgtServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+                    }
+                    continue;
+                }
+                trustedAndroidApps.add(trustedAndroidApp);
+            }
+        } catch (IdentityApplicationManagementException e) {
+            throw new TrustedAppMgtException("Error occurred while retrieving trusted apps for " +
+                    "Android platform.", e);
+        }
+        return trustedAndroidApps;
+    }
+
+    @Override
+    public List<TrustedIosApp> getTrustedIosApps() throws TrustedAppMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving trusted apps for iOS platform.");
+        }
+        List<TrustedIosApp> trustedIosApps = new ArrayList<>();
+        try {
+            List<TrustedApp> trustedAppSet =
+                    TrustedAppMgtDataHolder.getInstance().getApplicationManagementService().
+                            getTrustedApps(PlatformType.IOS);
+            for (TrustedApp trustedApp : trustedAppSet) {
+                TrustedIosApp trustedIosApp = new TrustedIosApp();
+                trustedIosApp.setAppId(trustedApp.getAppIdentifier());
+                try {
+                    trustedIosApp.setPermissions(resolveAppPermissions(trustedApp));
+                } catch (TrustedAppMgtException e) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.");
+                    }
+                    continue;
+                }
+                trustedIosApps.add(trustedIosApp);
+            }
+        } catch (IdentityApplicationManagementException e) {
+            throw new TrustedAppMgtException("Error occurred while retrieving trusted apps for " +
+                    "iOS platform.", e);
+        }
+        return trustedIosApps;
+    }
+
+    private String[] resolveThumbprints(TrustedApp trustedApp) throws TrustedAppMgtException {
+
+        String[] thumbprints = trustedApp.getThumbprints();
+        if (thumbprints == null || thumbprints.length == 0) {
+            throw new TrustedAppMgtException(String.format("No thumbprints found for the app: %s. ",
+                    trustedApp.getAppIdentifier()));
+        }
+        return thumbprints;
+    }
+
+    private Set<String> resolveAppPermissions(TrustedApp trustedApp) throws TrustedAppMgtException {
+
+        Set<String> appPermissions = new HashSet<>();
+        if (PlatformType.ANDROID.equals(trustedApp.getPlatformType())) {
+            if (trustedApp.getIsFIDOTrusted()) {
+                appPermissions.add(ANDROID_CREDENTIAL_PERMISSION);
+                appPermissions.add(ANDROID_HANDLE_URLS_PERMISSION);
+            }
+        } else if (PlatformType.IOS.equals(trustedApp.getPlatformType()) && trustedApp.getIsFIDOTrusted()) {
+            appPermissions.add(IOS_CREDENTIAL_PERMISSION);
+        }
+        if (appPermissions.isEmpty()) {
```

Comment:
instead of throwing an exception lets return an empty array and lets the calling component deal with the empty array. Alternatively can use java optional when returning.